src/components/Process/Create/index.tsx

```diff
@@ -203,6 +220,76 @@ const TemplateConfigs: Record<TemplateTypes, TemplateConfig> = {
   },
 }
 
+export const useConfirmOnNavigate = ({
+  isDirty,
+  isSubmitting,
+  isSubmitSuccessful,
+  onOpen,
+  onClose,
+}: ConfirmOnNavigateOptions) => {
+  const shouldBlock = isDirty && !isSubmitting && !isSubmitSuccessful
+  const blocker = useBlocker(shouldBlock)
+
+  const isOpenRef = useRef(false)
+  const isProceedingRef = useRef(false)
+
+  const currentPath = `${window.location.pathname}${window.location.search}${window.location.hash}`
+  const nextPath = blocker.location ? createPath(blocker.location) : null
+  const isSamePath = nextPath === currentPath
+
+  useEffect(() => {
+    if (!shouldBlock) {
+      if (isOpenRef.current) {
+        isOpenRef.current = false
+        onClose()
+      }
+      if (blocker.state === 'blocked') {
+        blocker.reset()
+      }
+      return
+    }
+
+    if (blocker.state === 'blocked' && !isOpenRef.current && !isProceedingRef.current) {
+      isOpenRef.current = true
+      onOpen()
+    }
+
+    if (blocker.state === 'unblocked' && isOpenRef.current) {
+      isOpenRef.current = false
+      onClose()
+    }
+  }, [blocker.state, shouldBlock, onOpen, onClose])
+
+  const closeAll = () => {
+    isProceedingRef.current = false
+    isOpenRef.current = false
+    onClose()
+  }
+
+  const cancel = () => {
+    closeAll()
+    blocker.reset()
+  }
+
+  const proceed = () => {
+    isProceedingRef.current = true
+    closeAll()
+    blocker.proceed()
+    // limpieza defensiva en el siguiente tick
+    setTimeout(() => {
+      isProceedingRef.current = false
+      blocker.reset()
+    }, 0)
+  }
```

Comment:
I think you should:
- ensure this method does not save the form, since right now the onfocus triggered by the form is storing in local storage and here we're not resetting it
- change the name of the method to something like `cancel` or even reuse the reset method when cancelling